src/test/java/lambda/part1/exercise/Exercise1.java

```diff
@@ -0,0 +1,114 @@
+package lambda.part1.exercise;
+
+import com.google.common.base.Predicate;
+import com.google.common.collect.FluentIterable;
+import java.util.Comparator;
+import lambda.data.Person;
+import java.util.Arrays;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertArrayEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class Exercise1 {
+
+    @Test
+    public void sortPersonsByAgeUsingArraysSortComparator() {
+        Person[] persons = getPersons();
+
+        // TODO использовать Arrays.sort
+
+        Comparator<Person> comparatorPersonByAdge = new Comparator<Person>() {
+            @Override
+            public int compare(Person o1, Person o2) {
+                return Integer.compare(o1.getAge(),o2.getAge());
+            }
+        };
+
+        Arrays.sort(persons, comparatorPersonByAdge);
+
+        assertArrayEquals(new Person[]{
+            new Person("Иван", "Мельников", 20),
+            new Person("Николай", "Зимов", 30),
+            new Person("Алексей", "Доренко", 40),
+            new Person("Артем", "Зимов", 45)
+        }, persons);
+    }
+
+    @Test
+    public void sortPersonsByAgeUsingArraysSortAnonymousComparator() {
+        Person[] persons = getPersons();
+
+        // TODO использовать Arrays.sort
+        Arrays.sort(persons, new Comparator<Person>() {
+            @Override
+            public int compare(Person o1, Person o2) {
+                return Integer.compare(o1.getAge(),o2.getAge());
+            }
+        });
+
+        assertArrayEquals(new Person[]{
+            new Person("Иван", "Мельников", 20),
+            new Person("Николай", "Зимов", 30),
+            new Person("Алексей", "Доренко", 40),
+            new Person("Артем", "Зимов", 45)
+        }, persons);
+    }
+
+    @Test
+    public void sortPersonsByLastNameThenFirstNameUsingArraysSortAnonymousComparator() {
+        Person[] persons = getPersons();
+
+        // TODO использовать Arrays.sort
+        Arrays.sort(persons, new Comparator<Person>() {
+            @Override
+            public int compare(Person o1, Person o2) {
+                return o1.getLastName().compareTo(o2.getLastName())==0
+                    ?o1.getFirstName().compareTo(o2.getFirstName())
+                    :o1.getLastName().compareTo(o2.getLastName());
+            }
+        });
+
+        assertArrayEquals(new Person[]{
+            new Person("Алексей", "Доренко", 40),
+            new Person("Артем", "Зимов", 45),
+            new Person("Николай", "Зимов", 30),
+            new Person("Иван", "Мельников", 20)
+        }, persons);
+    }
+
+    @Test
+    public void findFirstWithAge30UsingGuavaPredicate() {
+        List<Person> persons = Arrays.asList(getPersons());
+
+        // TODO использовать FluentIterable
+        Person person = null;
+        person = FluentIterable.from(persons)
+            .filter(value->value.getAge()==30).first().get();
```

Comment:
You should use Guave Predicate here, not lambda

Reply:
Done.